contract/shiden/contracts/AstarBaseV4.sol

```diff
@@ -169,6 +171,12 @@ contract AstarBaseV4
             // register to avoid check in external astarbase next time
             registerExecute(evmAddress, ss58PublicKey);
         }
+        else{
+            // if ss58PublicKey32 was 0, it will be encoded as 0x000...000
+            // this will cause chack in isRegister to show len(0)>0 
+            delete ss58PublicKey;
```

Comment:
We cannot modify blockchain state in isRegistered function. It should remain a view function

Reply:
you are right. But the reason that this method needs signing is in call to register()
this delete is for the local variable

Reply:
Need a view function to show user is registered or not.

Reply:
done